### PR TITLE
Document spec and status downward API fields  only available as environment variables

### PR DESCRIPTION
**Pull request by Kartik494**
Fixes: #38361 

This PR Modify spec and status downward API fields to  only available as environment variables not as files.

### CODE REVIEW

**Comment by Kartik494** [comment]
Hi @tengqm PTAL!
/sig node

File: content/en/docs/concepts/workloads/pods/downward-api.md [hunk 1]
```diff
@@ -63,6 +63,8 @@ via either mechanism are:
 `metadata.labels['<KEY>']`
 : the text value of the pod's {{< glossary_tooltip text="label" term_id="label" >}} named `<KEY>` (for example, `metadata.labels['mylabel']`)
 
+The following information is available through environment variables **but not as a downwardAPI volume fieldRef**:
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/pods/downward-api.md:66]:
```suggestion
The following information is available through environment variables
**but not as a downwardAPI volume `fieldRef`**:
```

File: content/en/docs/concepts/workloads/pods/downward-api.md [hunk 2]
```diff
@@ -75,8 +77,7 @@ via either mechanism are:
 `status.podIP`
 : the pod's primary IP address (usually, its IPv4 address)
 
-In addition, the following information is available through
-a `downwardAPI` volume `fieldRef`, but **not as environment variables**:
+The following information is available through a `downwardAPI` volume `fieldRef`, **but not as environment variables**:
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/pods/downward-api.md:80]:
```suggestion
The following information is available through a `downwardAPI` volume `fieldRef`,
**but not as environment variables**:
```

File: content/en/docs/concepts/workloads/pods/downward-api.md (hunk 1 above)
Comment (Kartik494) [reply content/en/docs/concepts/workloads/pods/downward-api.md]: This is done.

File: content/en/docs/concepts/workloads/pods/downward-api.md (hunk 2 above)
Comment (Kartik494) [reply content/en/docs/concepts/workloads/pods/downward-api.md]: I have addressed the changes as you suggested. PTAL!!

File: content/en/docs/concepts/workloads/pods/downward-api.md [hunk 3]
```diff
@@ -75,8 +78,9 @@ via either mechanism are:
 `status.podIP`
 : the pod's primary IP address (usually, its IPv4 address)
 
-In addition, the following information is available through
-a `downwardAPI` volume `fieldRef`, but **not as environment variables**:
+The following information is available through a `downwardAPI` volume 
+`fieldRef`,
+**but not as environment variables**:
```
Comment (tengqm) [inline content/en/docs/concepts/workloads/pods/downward-api.md:83]: Nit: This extra line wrapping is not needed.

Comment (Kartik494) [reply content/en/docs/concepts/workloads/pods/downward-api.md]: Addressed this change. PTAL!!

**Comment by reylejano** [comment]
Preview of updated page:
https://deploy-preview-39072--kubernetes-io-main-staging.netlify.app/docs/concepts/workloads/pods/downward-api/

/lgtm

**Pull request merged**